### PR TITLE
[Commands] Fix swift compiler paths

### PR DESCRIPTION
**Pull request by aciidb0mb3r**

### CODE REVIEW

**Comment by aciidb0mb3r** [comment]
@swift-ci please smoke test

**Comment by jrose-apple** [comment]
Worth mentioning that / when the behavior changed compared to previous SwiftPM releases?

**Comment by aciidb0mb3r** [comment]
I don't think anybody relies on this except SwiftPM itself.

File: Documentation/Development.md [hunk 1]
```diff
@@ -134,19 +134,23 @@ $ Utilities/docker-utils run bash # to run an interactive bash shell in the cont
...
+environment variables to control which compiler to use for these operations.
 
-You can use a custom Swift compiler for compiling the package sources using the
-environment variable `SWIFT_EXEC_COMPILE_ONLY`. The package manager will use the
-default Swift compiler to parse the manifest files but use the compiler in
-`SWIFT_EXEC_COMPILE_ONLY` to compile the source files. This is mostly useful for
-Swift compiler developers when they want to use a debug compiler with SwiftPM.
+`SWIFT_EXEC_MANIFEST`: This variable controls which compiler to use for parsing
+Package.swift manifest files. The lookup order for manifest compiler is:
+SWIFT_EXEC_MANIFEST, swiftc adjacent to swiftpm binaries, SWIFT_EXEC
```
Comment (neonichu) [inline Documentation/Development.md]: Is the trailing `SWIFT_EXEC` actually right here? From implementation it looks like not `SWIFT_EXEC_MANIFEST ?? resolvedBinDirCompiler`.

Comment (aciidb0mb3r) [reply Documentation/Development.md]: Yeah, this is correct I thnk. See this if clause https://github.com/apple/swift-package-manager/pull/1444/files#diff-8c6d7a38675f213e4370feadb3a37ed7R87

Comment (neonichu) [reply Documentation/Development.md]: Ah right, I missed that somehow 🙈

**Pull request merged**